### PR TITLE
initial version of mintmaker service

### PR DESCRIPTION
**Pull request by scoheb**
- This is just a simple skeleton controller such
  that we can get mintmaker integrated.

### CODE REVIEW

File: components/mintmaker/base/allow-argocd-to-manage.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
```
Comment (gbenhaim) [inline components/mintmaker/base/allow-argocd-to-manage.yaml:2]: this is not needed anymore

**Comment by gbenhaim** [comment]
@scoheb before deploying it, I asked the mintmaker team for a technical overview. I would like to understand how this service is going to work and what it will require from the cluster.

File: argo-cd-apps/overlays/development/kustomization.yaml [hunk 2]
```diff
@@ -101,6 +101,11 @@ patches:
       kind: ApplicationSet
       version: v1alpha1
       name: release
+  - path: development-overlay-patch.yaml
+    target:
+      kind: ApplicationSet
+      version: v1alpha1
+      name: mintmaker
```
Comment (hugares) [inline argo-cd-apps/overlays/development/kustomization.yaml:108]: this patch is setting the environment to `development` in the ApplicationSet which is what we want. You need to do the same in but using `production-overlay-patch.yaml`:
- argo-cd-apps/overlays/konflux-public-production/kustomization.yaml
- argo-cd-apps/overlays/production-downstream/kustomization.yaml

File: components/mintmaker/development/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../base
+  - https://github.com/konflux-ci/mintmaker/config/default?ref=9e3dd3bfec43fbbae2d2dfb931f3f3cb082ae9af
```
Comment (hugares) [inline components/mintmaker/development/kustomization.yaml:5]: The manifest brought by this revision do have kube linter issues, this needs to be fixed as we have a check for that, which is failing. Pasting the issues here so you do not have to drill down in the check logs:
```
<standard input>: (object: mintmaker/mintmaker-controller-manager apps/v1, Kind=Deployment) container "kube-rbac-proxy" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in the container securityContext.)

<standard input>: (object: mintmaker/mintmaker-controller-manager apps/v1, Kind=Deployment) container "manager" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in the container securityContext.)
```

**Review by hugares — CHANGES_REQUESTED** [review]

**Comment by ifireball** [comment]
Can we have some context information added to this PR? The linked repo only seems to contain very basic auto-generated code and there is not much information in the README or any of the commits. We don't have any link here to any ADRs or Jira issues. "mintmaker" isn't a Google friendly term either.

**Comment by gnaponie** [comment]
> Can we have some context information added to this PR? The linked repo only seems to contain very basic auto-generated code and there is not much information in the README or any of the commits. We don't have any link here to any ADRs or Jira issues. "mintmaker" isn't a Google friendly term either.

Hello,
perhaps I can provide some context here.
My team is working on providing the Renovate service in Konflux to update dependencies and for future CVE remediation. And for this solution the name "Mintmaker" was chosen. It will substitute logically what Freshmaker does in the classical pipeline (here comes the name Mintmaker, in Freshmaker's honor).
The request is coming from Brian Cook and Ralph Bean.

File: components/mintmaker/base/mintmaker-team.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mintmaker-maintainers
+  namespace: mintmaker-service
```
Comment (gnaponie) [inline components/mintmaker/base/mintmaker-team.yaml:5]: Is this intended? Or should it be just `mintmaker`? I see in all other places it is `mintmaker` and not `mintmaker-service`.

**Comment by ifireball** [comment]
> > Can we have some context information added to this PR? The linked repo only seems to contain very basic auto-generated code and there is not much information in the README or any of the commits. We don't have any link here to any ADRs or Jira issues. "mintmaker" isn't a Google friendly term either.
> 
> Hello, perhaps I can provide some context here. My team is working on providing the Renovate service in Konflux to update dependencies and for future CVE remediation. And for this solution the name "Mintmaker" was chosen. It will substitute logically what Freshmaker does in the classical pipeline (here comes the name Mintmaker, in Freshmaker's honor). The request is coming from Brian Cook and Ralph Bean.

@gnaponie thanks! Can you provide a link to a Jira issue tracking this work? Also ironing-out the technical details via the ADR process would be much appreciated, if it was not done yet.

**Comment by gnaponie** [comment]
> @gnaponie thanks! Can you provide a link to a Jira issue tracking this work? Also ironing-out the technical details via the ADR process would be much appreciated, if it was not done yet.

Sure! Here's the jira issue: https://issues.redhat.com/browse/CLOUDWF-10108 

We have a document, but that perhaps it is a bit outdated. User stories and requirements in that document are in line with the request, but the actual technical details and proposal plan is a bit different. Would it be ok if I send that?

**Comment by gnaponie** [comment]
So I've talked to @scoheb, and actually we've decided we're going to close this PR in favor of mine, which can be found here: https://github.com/redhat-appstudio/infra-deployments/pull/3798 
This new PR includes Scott's commit.
@ifireball I don't think I have permission to close this PR, could you do it and we can continue our conversation in the other PR? Thank you!

**Comment by scoheb** [comment]
closing in favour of [3798](https://github.com/redhat-appstudio/infra-deployments/pull/3798)

**Pull request closed without merging**